src/Workspaces/Core/Portable/Utilities/NullableHelpers/TypeSymbolWithNullableAnnotation.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Globalization;
+using System.Text;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class NullableExtensions
+    {
+        [DebuggerDisplay("{WrappedSymbol,nq}, Nullability = {Nullability,nq}")]
+        private abstract class TypeSymbolWithNullableAnnotation : ITypeSymbol
+        {
+            internal ITypeSymbol WrappedSymbol { get; }
+            internal NullableAnnotation Nullability { get; }
+
+            protected TypeSymbolWithNullableAnnotation(ITypeSymbol wrappedSymbol, NullableAnnotation nullability)
+            {
+                Debug.Assert(!(wrappedSymbol is TypeSymbolWithNullableAnnotation));
+
+                WrappedSymbol = wrappedSymbol;
+                Nullability = nullability;
+            }
+
+            public bool Equals(ISymbol other)
+            {
+                if (other is TypeSymbolWithNullableAnnotation otherWrappingSymbol)
+                {
+                    return this.Nullability == otherWrappingSymbol.Nullability &&
+                           this.WrappedSymbol.Equals(otherWrappingSymbol.WrappedSymbol);
```

Comment:
this seems divergent with the compiler's view on what .Equals should do (namely that it won't compare nullability).

Reply:
Once we have the analyzer in place and we move everything to explicit comparers then we'll update accordingly -- probably by making this throw. :smile: This is more or less to maintain compat for things that we haven't looked at.